validator.js

```diff
@@ -202,6 +205,11 @@
         if (query && /\s/.test(query)) {
             return false;
         }
+
+        if (options.allow_protocol_relative_urls && protocolRelativeURLPrefix.test(url)) {
+            url = protocolRelativeURLPrefix.exec(url)[1];
```

Comment:
We thought it read a little clearer to have the regex named and to reuse it rather than introduce a second way of describing the url matcher. That said I'm happy to switch it for the substr style if you prefer?  We'll fix http:////foobar.com now